packages/commercetools/theme/components/Auth/RegisterForm.vue

```diff
@@ -0,0 +1,205 @@
+        <ValidationProvider rules="required" v-slot="{ errors }">
+          <SfInput
+            data-cy="login-input_password"
+            v-model="form.password"
+            :valid="!errors[0]"
+            :errorMessage="$t(errors[0])"
+            name="password"
+            :label="$t('Password')"
+            type="password"
+            class="form__element"
+          />
+        </ValidationProvider>
+        <ValidationProvider :rules="{ required: { allowFalse: false } }" v-slot="{ errors }">
+          <SfCheckbox
+            v-model="createAccount"
+            :valid="!errors[0]"
+            :errorMessage="$t(errors[0])"
+            name="create-account"
+            :label="$t('I want to create an account')"
+            class="form__element"
+          />
+        </ValidationProvider>
+        <SfButton
+          data-cy="login-btn_submit"
+          type="submit"
+          class="sf-button--full-width form__button"
+          :disabled="loading"
+        >
+          <SfLoader :class="{ loader: loading }" :loading="loading">
+            <div>{{ $t('Create an account') }}</div>
+          </SfLoader>
+        </SfButton>
+      </form>
+    </ValidationObserver>
+    <div class="action">
+      {{ $t('or') }}
+      <SfButton data-cy="login-btn_login-into-account" class="sf-button--text" @click="switchAuthModal('login')">
+        {{ $t('login in to your account') }}
+      </SfButton>
+    </div>
+  </div>
+</template>
+
+<script>
+import { extend, ValidationObserver, ValidationProvider } from 'vee-validate';
+import { SfButton, SfCheckbox, SfInput, SfLoader, SfAlert } from '@storefront-ui/vue';
+import { email, required } from 'vee-validate/dist/rules';
+import { ref } from '@vue/composition-api';
+import { useUser } from '@vue-storefront/commercetools';
+import { useUiState, useUiNotification } from '~/composables';
+
+extend('email', {
+  ...email,
+  message: 'Invalid email'
+});
+
+extend('required', {
+  ...required,
+  message: 'This field is required'
+});
+
+export default {
+  name: 'RegisterForm',
+  components: {
+    ValidationProvider,
+    ValidationObserver,
+    SfButton,
+    SfCheckbox,
+    SfInput,
+    SfLoader,
+    SfAlert
+  },
+  setup(_, context) {
+    const { register, loading, error } = useUser();
+    const { toggleAuthModal, switchAuthModal } = useUiState();
+    const { send } = useUiNotification();
+    const form = ref({});
+    const createAccount = ref(false);
+    const { $i18n } = context.root;
+    const registerError = ref(null);
+
+    const handleError = () => {
+      registerError.value = error.value.register;
+      if (!registerError.value) {
+        send({
+          type: 'success',
+          message: $i18n.t('Successfully created a new account')
+        });
+        toggleAuthModal();
+        return;
+      }
+
+      if (registerError.value && registerError.value.code === 'DuplicateField' && registerError.value.message.search(form.value.email) !== -1) {
```

Comment:
jeezzz, why that's sooo long? :D is it readable? can you create a function for checking this or even a computed property?